### PR TITLE
#161779777 implement JWT authentication 

### PR DESCRIPTION
**Pull request by Oluwasegun-AA**
#### What does this PR do?
This PR Implements authentication using JSON Web Token (JWT)

#### Description of Task to be completed?
* add authentication.js file
* implement the `getToken()` function
* encrypt token after its generated
* implement `verifyToken()` function
* decrypt token before verification
* add test cases for authentication

#### How should this be manually tested?
* clone the repository  ` https://github.com/andela/helios-ah-backend.git `
* cd to the  ` helios-ah-backend ` folder
* checkout the `develop` branch
* run `npm i` on the bash terminal
* run `npm run test` on the terminal

#### What are the relevant pivotal tracker stories?
#161779777

#### Any background context you want to add?
N/A

#### Screenshots
screen shot 2018-12-04 at 1 35 11 pm

### CODE REVIEW

File: src/utilities/authentication.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+function scramble(a) {
+  return a.split('').reverse().join('');
+}
+
+/**
+ * Unscrambles string data
+ * @param {*} a - input string of scrambled data
+ * @returns {output} - unscrambled data
+ */
+function unscramble(a) {
```
Comment (kingisaac95) [inline src/utilities/authentication.js]: Remove this function. It is redundant

File: src/utilities/authentication.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import dotenv from 'dotenv';
+
+dotenv.config();
+
+/**
+ * scrambles string data
+ * @param {*} a - input string data
+ * @returns {output} - scrambled data
+ */
+function scramble(a) {
```
Comment (kingisaac95) [inline src/utilities/authentication.js]: Give this function a more generic name. Eg. `reverseToken`

File: test/authenticationTest.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    expect(token).to.be.a('String');
+  });
+});
+
+describe('scambled token should be unscrambled and Validated', () => {
+  it('it should unscramble token and validate', async() => {
+    const token = await getToken(data);
+    const user = await verifyToken(token);
+    console.log(user)
+    console.log(user.id)
```
Comment (kingisaac95) [inline test/authenticationTest.js]: Remove console statements

Comment (Oluwasegun-AA) [reply test/authenticationTest.js]: Console statements removed.

File: test/authenticationTest.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+describe('scambled token should be unscrambled and Validated', () => {
+  it('it should unscramble token and validate', async() => {
+    const token = await getToken(data);
+    const user = await verifyToken(token);
+    expect(user.id).to.be.equal(2);
+    expect(user.role).to.be.equal(1);
+    expect(user.userName).to.be.equal('helios');
+    expect(user).to.be.a('object');
+  });
+});
```
Comment (andrewinsoul) [inline test/authenticationTest.js]: Newline is required here

Comment (Oluwasegun-AA) [reply test/authenticationTest.js]: Thanks for the feedback. It has been fixed

File: src/utilities/authentication.js (hunk 2 above)
Comment (kingisaac95) [inline src/utilities/authentication.js]: Make the variable name more descriptive. Eg.

`const reverseToken = (token) => token.split('').reverse().join('');`

The above is basically what I'm asking for

**Review by kingisaac95 — APPROVED** [review]

**Pull request merged**